symphony-bdk-core/src/main/java/com/symphony/bdk/core/retry/RetryWithRecoveryBuilder.java

```diff
@@ -38,78 +55,73 @@
    * @return a copy of the builder passed as parameter.
    */
   public static <T> RetryWithRecoveryBuilder<T> from(RetryWithRecoveryBuilder<?> from) {
-    RetryWithRecoveryBuilder<T> copy = new RetryWithRecoveryBuilder();
-    copy.name = from.name;
-    copy.address = from.address;
-    copy.retryConfig = from.retryConfig;
-    copy.retryOnExceptionPredicate = from.retryOnExceptionPredicate;
-    copy.ignoreException = from.ignoreException;
+    final RetryWithRecoveryBuilder<T> copy = copyWithoutRecoveryStrategies(from);
     copy.recoveryStrategies = new ArrayList<>(from.recoveryStrategies);
-
     return copy;
   }
 
   public static <T> RetryWithRecoveryBuilder<T> copyWithoutRecoveryStrategies(RetryWithRecoveryBuilder<?> from) {
-    RetryWithRecoveryBuilder<T> copy = new RetryWithRecoveryBuilder();
+    final RetryWithRecoveryBuilder<T> copy = new RetryWithRecoveryBuilder<>();
     copy.name = from.name;
     copy.address = from.address;
     copy.retryConfig = from.retryConfig;
     copy.retryOnExceptionPredicate = from.retryOnExceptionPredicate;
     copy.ignoreException = from.ignoreException;
-
     return copy;
   }
 
   /**
    * Checks if a throwable is a network issue or a {@link ApiException} minor error.
-   * This is the default function used in {@link RetryWithRecovery}
-   * to check if a given exception thrown should lead to a retry.
+   * <p>This is the default function used in {@link RetryWithRecovery} to check if a given exception thrown should lead
+   * to a retry or not.
    *
    * @param t the throwable to be checked.
-   * @return true if passed throwable is either a {@link SocketTimeoutException} or {@link ConnectException}
-   * or if it is a {@link ApiException} which {@link ApiException#isServerError()}
-   * or {@link ApiException#isUnauthorized()} or {@link ApiException#isTooManyRequestsError()}.
+   * @return true if a given throwable is either a {@link SocketTimeoutException}, a {@link SocketException},
+   * a {@link ConnectException}, a {@link UnknownHostException} or if it's am {@link ApiException} that verifies
+   * {@link ApiException#isServerError()}, {@link ApiException#isUnauthorized()} or {@link ApiException#isTooManyRequestsError()}.
    */
   public static boolean isNetworkOrMinorError(Throwable t) {
     if (t instanceof ApiException) {
       ApiException apiException = (ApiException) t;
       return apiException.isServerError() || apiException.isUnauthorized() || apiException.isTooManyRequestsError();
     }
-    return t.getCause() instanceof SocketTimeoutException || t.getCause() instanceof ConnectException;
+    // keep the datafeed loop running for errors that might be temporary network errors
+    return rootCauseCouldBe(t, SocketException.class)
```

Comment:
in the end aren't we better off doing the reverse? do we know the use cases where we don't want to retry?